plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1418,17 +1419,24 @@ public void renameColumn(ConnectorSession session, ConnectorTableHandle tableHan
     private ConnectorTableMetadata getTableMetadata(ConnectorSession session, SchemaTableName table)
     {
         Table icebergTable = catalog.loadTable(session, table);
+        List<ColumnMetadata> columns = getColumnMetadatas(icebergTable.schema());
 
-        ImmutableList.Builder<ColumnMetadata> columns = ImmutableList.builder();
-        columns.addAll(getColumnMetadatas(icebergTable));
-        columns.add(pathColumnMetadata());
+        return new ConnectorTableMetadata(table, columns, getIcebergTableProperties(icebergTable), getTableComment(icebergTable));
+    }
+
+    private ConnectorTableMetadata getTableMetadata(ConnectorSession session, SchemaTableName table, Schema schema)
+    {
+        Table icebergTable = catalog.loadTable(session, table);
+        List<ColumnMetadata> columns = getColumnMetadatas(schema);
 
-        return new ConnectorTableMetadata(table, columns.build(), getIcebergTableProperties(icebergTable), getTableComment(icebergTable));
+        return new ConnectorTableMetadata(table, columns, getIcebergTableProperties(icebergTable), getTableComment(icebergTable));
```

Comment:
Do table comments / properties also have a history we should be looking through during TT?